infra/terraform/modules/data_access/lambda_users.tf

```diff
@@ -0,0 +1,155 @@
+# Zip the lambda function before the actual deploy
+data "archive_file" "users_zip" {
+    type        = "zip"
+    source_dir  = "${path.module}/users"
+    output_path = "/tmp/users.zip"
+}
+
+# Lambda function adds an IAM role given its username
+resource "aws_lambda_function" "create_user_role" {
+    description = "Adds an IAM role for the given user"
+    filename = "/tmp/users.zip"
+    source_code_hash = "${data.archive_file.users_zip.output_base64sha256}"
+    function_name = "${var.env}_create_user_role"
+    role = "${aws_iam_role.create_user_role.arn}"
+    handler = "users.create_user_role"
+    runtime = "python3.6"
+    timeout = 10
+    depends_on = ["data.archive_file.users_zip"]
+    environment {
+        variables = {
+            STAGE = "${var.env}",
+            SAML_PROVIDER_ARN = "arn:aws:iam::${var.account_id}:saml-provider/auth0",
+        }
+    }
+}
+
+# Role assumed by the 'create_user_role' lambda function
+resource "aws_iam_role" "create_user_role" {
```

Comment:
I did this way so that for example the `create_role` lambda function doesn't have the policy to perform `iam:DeleteRole`.

I guess these two lambda function are closely related so we may go for a single role with the policies to create/delete an IAM role and also detach policies, which is a bit simpler.